acstools/acszpt.py

```diff
@@ -0,0 +1,329 @@
+
+    Parameters
+    ----------
+    date : str
+        Input date in the following ISO format, YYYY-MM-DD
+    detector : str
+        One of the three channels on ACS: HRC, SBC, or WFC
+    filt : str, optional
+        One of valid filters for the chosen detector
+            * HRC:
+                 F220W, F250W, F330W,
+                 F344N, F435W, F475W,
+                 F502N, F550M, F555W,
+                 F606W, F625W, F658N, F660N,
+                 F775W, F814W, F850LP, F892N
+
+            * WFC:
+                 F435W, F475W,
+                 F502N, F550M, F555W,
+                 F606W, F625W, F658N, F660N,
+                 F775W, F814W, F850LP, F892N
+            * SBC:
+                F115LP, F122M, F125LP,
+                F140LP, F150LP, F165LP
+
+    Attributes
+    ----------
+    date : str
+        **Attribute** for storing the user specified date.
+    detector : str
+        **Attribute** for storing the user specified detector.
+    filt : str, None
+        **Attribute** for storing the user specified filter (if supplied).
+    zpt_table : astropy.table.Table
+        **Attribute** for storing the results returned by the ACS Zeropoint Calculator
+    """
+
+    def __init__(self, date, detector, filt=None):
+
+        # Set the public attributes
+        self.date = date
+        self.detector = detector.upper()
+        self.filt = filt
+
+        self.valid_filters = {
+            'WFC': ['F435W', 'F475W', 'F502N', 'F550M',
+                    'F555W', 'F606W', 'F625W', 'F658N',
+                    'F660N', 'F775W', 'F814W', 'F850LP','F892N'],
+
+            'HRC': ['F220W', 'F250W', 'F330W', 'F344N',
+                    'F435W', 'F475W', 'F502N', 'F550M',
+                    'F555W', 'F606W', 'F625W', 'F658N',
+                    'F660N', 'F775W', 'F814W', 'F850LP','F892N'],
+
+            'SBC': ['F115LP', 'F122M', 'F125LP',
+                   'F140LP', 'F150LP', 'F165LP']
+        }
+        self.zpt_table = None
+
+        # Set the private attributes
+        if filt is None:
+            self._url = 'https://acszeropoints.stsci.edu/results_all/?date={}&' \
+                        'detector={}'.format(self.date, self.detector)
+        else:
+            self.filt = filt.upper()
+
+            self._url = 'https://acszeropoints.stsci.edu/results_single/' \
+                        '?date1={0}&detector={1}&{1}_filter={2}'. \
+                format(self.date, self.detector, self.filt)
+
+        self._zpts = OrderedDict()
+        self._response = None
+        self._failed = False
+
+    def _check_inputs(self):
+        """ Check the input to ensure they are valid
+
+        Returns
+        -------
+
+        """
+        valid_date = True
+        valid_detector = True
+        valid_filter = True
+        # Determine the submitted detector is valid
+        if self.detector not in ['HRC','SBC','WFC']:
+            print('Whoops! {} is not a valid detector option.'.
```

Comment:
Using `print` is not recommended. Please consider using Python logging system.

Also, is there a reason why not raise an error (e.g., `ValueError`) straight away? It would be easier to debug if traceback points out where the comparison is failing. In addition, checking behavior is different from `_check_date`, which does raise an error. I think this can be simplified and made more consistent.

Reply:
I've never used the logging system, any suggestions on where to find good examples of the implementation? 

The point of these checks is not for the user to debug issues with the code. It's for me to know whether or not they've instantiated the object with valid inputs. If any of the inputs are invalid, the user is informed of which inputs were invalid and what the acceptable inputs are. For example:

An incorrect detector is supplied but the date is correct:
```console 
In [6]: q = acszpt.Query(detector='UVIS', date='2004-02-01')                              

In [7]: q.fetch()                                                                         
Whoops! UVIS is not a valid detector option.
Please choose one of the following:
HRC
SBC
WFC
-------------------------------------------------------------------------------
Please fix the incorrect input(s)
```
An incorrect detector and date are supplied:
```console 
In [8]: q = acszpt.Query(detector='UVIS', date='1999-02-01')                              

In [9]: q.fetch()                                                                         
Whoops! UVIS is not a valid detector option.
Please choose one of the following:
HRC
SBC
WFC
-------------------------------------------------------------------------------
Whoops! The observation cannot occur before ACS was installed (2002-03-07)
-------------------------------------------------------------------------------
Please fix the incorrect input(s)
```
An incorrect date and filter are supplied with a valid detector:
```console
In [10]: q = acszpt.Query(detector='WFC', date='1999-02-01',filt='F140LP')                 

In [11]: q.fetch()                                                                         
Whoops! F140LP is not a valid filter for WFC
Please choose one of the following:
F435W
F475W
F502N
F550M
F555W
F606W
F625W
F658N
F660N
F775W
F814W
F850LP
F892N
-------------------------------------------------------------------------------
Whoops! The observation cannot occur before ACS was installed (2002-03-07)
-------------------------------------------------------------------------------
Please fix the incorrect input(s)
```
Finally, if the detector is invalid and a filter is supplied, the code will skip the filter checking because the detector is not valid and I am not going code every filter for every detector on HST into this. 
```console
In [12]: q = acszpt.Query(detector='UVIS', date='1999-02-01',filt='F140LP')                

In [13]: q.fetch()                                                                         
Whoops! UVIS is not a valid detector option.
Please choose one of the following:
HRC
SBC
WFC
-------------------------------------------------------------------------------
Whoops! The observation cannot occur before ACS was installed (2002-03-07)
-------------------------------------------------------------------------------
Please fix the incorrect input(s)
```

In light of that, I don't see the point of raising errors. I am all for simplification of this code, but enforcing consistency for the sake of consistency seems pointless. I can modify the `try/except` in `_date_check` to simply return a `str` instead of the ValueError if it's _truly_ necessary though.

Reply:
There is some basic logging in
https://github.com/spacetelescope/acstools/blob/88529aadf4ab0348ff0ad25913da14e65517f522/acstools/acs_destripe.py#L92-L94